cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriter.java

```diff
@@ -99,6 +101,9 @@ public DatasetSystemMetadataWriter(MetadataStore metadataStore,
     if (description != null) {
       properties.put(DESCRIPTION, description);
     }
+    if (dataset instanceof FileSet || dataset instanceof PartitionedFileSet) {
+      properties.put("explore.enabled", "true");
```

Comment:
You should put a check around line 105 like so:
`if (FileSetProperties.isExploreEnabled(datasetProperties)) {`
because not all FileSets are explorable. It depends on its dataset properties.